cloudygames/views.py

```diff
@@ -1,28 +1,99 @@
 from django.shortcuts import render
+from django.core import serializers
 
-from rest_framework import viewsets, generics, status
+from rest_framework import viewsets, generics, status, filters
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
 
 from cloudygames.serializers import GameSerializer, GameSessionSerializer, PlayerSaveDataSerializer
 from cloudygames.models import Game, GameSession, PlayerSaveData
 
+import django_filters
+import json
+
+
+######################## Filter ############################
+
+
+class GameFilter(django_filters.FilterSet):
+    users = django_filters.CharFilter(name='users__username')
+
+    class Meta:
+        model = Game
+        fields = ['id', 'name', 'publisher', 'users']
+        order_by = ['name']
+        read_only_fields = ('id',)
+
+class GameSessionFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = GameSession
+        fields = ['game', 'player']
+
+class PlayerSaveDataFilter(django_filters.FilterSet):
+    game = django_filters.CharFilter(name='game__id')
+    player = django_filters.CharFilter(name='player_username')
+
+    class Meta:
+        model = PlayerSaveData
+        fields = ['game', 'player']
+
+
+######################## ViewSet ############################
+
 class GameViewSet(viewsets.ModelViewSet):
+    queryset = Game.objects.all()
     serializer_class = GameSerializer
+    filter_class = GameFilter
 
     def get_queryset(self):
         is_owned = self.request.query_params.get('owned', 0)
```

Comment:
There is a problem using Django-filter, that is I can not access the request directly, and hence I can not know the current user (self.request.user). That's why I filter it in the view class instead.